### PR TITLE
Targets are first class objects; remove rake:db:migrate being the default command

### PR DESCRIPTION
**Pull request by apurvis**
I couldn't take it anymore, having the `@targets` and `@target` in a `Deploy` object.

1. Now each `Target` is a top level object with only the details about... the target.
1. A `Deploy` is composed of a `Target` plus the options that come from the command line - the "deploy" options if you will.
1. There is no more `base` configuration - the main `Configuration` object holds all the `base` config. 
1. There is no more `deploy` configuration - that is all handled by the main `Configuration` and by `config.targets=`

All specs are passing.  Feel free to comment, disapprove, or offer other ideas on how to make this code a little easier to work with.  @camertron @rfroetscher @matl33t

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,17 +11,17 @@ Broadside offers a simple command-line interface to perform deployments on ECS.
 
 ```ruby
 Broadside.configure do |config|
-  config.base.application = 'hello_world'
-  config.base.docker_image = 'lumoslabs/hello_world'
-  config.deploy.type = 'ecs'
+  config.application = 'hello_world'
+  config.docker_image = 'lumoslabs/hello_world'
```
Comment (apurvis) [inline README.md]: if we ever think we will want to be able to configure `docker_image` on a per target basis, now would be the time to do it.  i left it how it was.

**Comment by apurvis** [comment]
OK i updated this so that now it is basically 100% backwards compatible except for the default `rake db:migrate` thing, which i think needs to come out at some point. it's also been tested.

File: README.md (hunk 1 above)
Comment (apurvis) [inline README.md]: @dollschasingmen / @timchan-lumoslabs

**Comment by apurvis** [comment]
@matl33t  are you concerned at all about the fact that `rake db:migrate` is no longer the default / do you see any problems with that?

i'm going to take this puppy through another round of testing all the commands then i will merge and someone should release (i'm not sure i can release yet?)

also i think after this change the code is getting to a point where maybe we should do a blog post or something; it's much more readable and work withable as code on top of the functionality which has always been pretty awesome.

File: bin/broadside [hunk 2]
```diff
@@ -44,9 +46,9 @@ end
 
 desc 'Bootstrap your service and task definition from the configured definition.'
 command :bootstrap do |b|
-  subcmd.desc 'Docker tag for application container'
-  subcmd.arg_name 'TAG'
-  subcmd.flag [:tag]
+  b.desc 'Docker tag for application container'
+  b.arg_name 'TAG'
+  b.flag [:tag]
```
Comment (apurvis) [inline bin/broadside]: also i'm not sure how this ever worked before this change... there is no `subcmd` in scope here?  /shrug

**Comment by matl33t** [comment]
no issues with removing `db:migrate` - since this will come with a major version bump we can roll this out without breaking applications

**Pull request merged**